File: Part 2 - Regression/Section 4 - Simple Linear Regression/simple_linear_regression.py
# -*- coding: utf-8 -*-
"""
Created on Sat Feb 11 09:50:44 2017

@author: Nott
"""

#simple lenear regression

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('Salary_Data.csv')
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 1].values

# Splitting the dataset into the Training set and Test set
from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 1/3, random_state = 0)

# Feature Scaling
"""from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)
sc_y = StandardScaler()
y_train = sc_y.fit_transform(y_train)"""

#fitting simple linear regression to traing set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train,y_train)

#predicting test set result
y_pred = regressor.predict(X_test)

#visualising training set result
plt.scatter(X_train,y_train, color = 'red')
plt.plot(X_train,regressor.predict(X_train), color = 'blue')
plt.title('Simple Lenear Regression (salary vs experience (training set))')
plt.xlabel('experience')
plt.ylabel('salary')
plt.show()

#visualising training set result
plt.scatter(X_test,y_test, color = 'red')
plt.plot(X_train,regressor.predict(X_train), color = 'blue')
plt.title('Simple Lenear Regression (salary vs experience (test set))')
plt.xlabel('experience')
plt.ylabel('salary')
plt.show()

File: Part 2 - Regression/Section 4 - Simple Linear Regression/simple_linear_regression.R
#simple linear regression

# Importing the dataset
dataset = read.csv('Salary_Data.csv')
# dataset = dataset[, 2:3]

# Splitting the dataset into the Training set and Test set
# install.packages('caTools')
library(caTools)
set.seed(123)
split = sample.split(dataset$Salary, SplitRatio = 2/3)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

# Feature Scaling
# training_set[, 2:3] = scale(training_set[, 2:3])
# test_set[, 2:3] = scale(test_set[, 2:3])

#fitter linear regression to training set
regressor = lm(formula = Salary ~ YearsExperience,
               training_set)

#predicting test set results
y_pred = predict(regressor,newdata = test_set)

#visualising train set results
#install.packages('ggplot2')
library(ggplot2)
ggplot() + 
  geom_point(aes(x = training_set$YearsExperience, y = training_set$Salary),
             colour = 'red') +
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor,training_set)),
             colour = 'blue') +
  ggtitle('Salary vs Experience (training set)') + 
  xlab('year of experience') +
  ylab('saraly')

#visualising test set results
ggplot() + 
  geom_point(aes(x = test_set$YearsExperience, y = test_set$Salary),
             colour = 'red') +
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor,training_set)),
            colour = 'blue') +
  ggtitle('Salary vs Experience (test set)') + 
  xlab('year of experience') +
  ylab('saraly')



